zebra-chain/src/parameters/network.rs

```diff
@@ -53,21 +50,109 @@ const ZIP_212_GRACE_PERIOD_DURATION: HeightDiff = 32_256;
 
 /// An enum describing the possible network choices.
 #[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Hash, Serialize, Deserialize)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
 pub enum Network {
     /// The production mainnet.
     #[default]
     Mainnet,
 
     /// The oldest public test network.
-    Testnet,
+    Testnet(#[serde(skip)] TestnetParameters),
+}
+
+/// Configures testnet network parameters to use instead of default values.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Hash, Serialize)]
+pub struct TestnetParameters(Option<NetworkParameters>);
```

Comment:
I don't think it's clear what `TestnetParameters(None)` means, and it makes the code more confusing and complex.

Instead, I suggest using a method that creates a testnet with the default parameters, wherever you would have used `None`.

This also improves test coverage, because we're using the same codepath for the standard testnet and alternate testnets.